packages/api/src/utils/db-client.js

```diff
@@ -10,6 +11,165 @@ export class DBClient {
         Accept: '*/*',
       },
     })
+    this._clientNftStorage = new PostgrestClient(endpoint, {
+      headers: {
+        Authorization: `Bearer ${token}`,
+        Accept: '*/*',
+      },
+      schema: 'nftstorage',
+    })
+  }
+
+  /**
+   * @param {Object} permaCache
+   * @param {number} permaCache.userId
+   * @param {string} permaCache.url
+   * @param {number} permaCache.size
+   * @param {string} permaCache.insertedAt
+   */
+  async createPermaCache(permaCache) {
+    const { data, error, status } = await this._client
+      .from('perma_cache')
+      .insert({
+        user_id: permaCache.userId,
+        url: permaCache.url,
+        size: permaCache.size,
+        inserted_at: permaCache.insertedAt,
+      })
+      .single()
+
+    if (error) {
+      if (status === HTTP_STATUS_CONFLICT) {
+        throw new ConstraintError({
+          message: 'URL already found for user',
+          status: HTTP_STATUS_CONFLICT,
```

Comment:
Do we ever need to create a `ConstraintError` with a HTTP status other than 409?